haas/api.py

```diff
@@ -423,6 +429,39 @@ def node_detach_network(node, nic, network):
     return '', 202
 
 
+@rest_call('PUT', '/node/<node>/metadata', Schema({
+    'node': basestring, 'value': basestring
+}))
+def node_register_metadata(node, label, value):
```

Comment:
You don't have a way to pass in the `label` parameter. Per my other comment, I would add it to the path. You'll also have to put it in the schema.